aws/sqs_event_manager/sqs_event_manager/app.py

```diff
@@ -0,0 +1,133 @@
+from sqs_event_manager.message import AWSSNSMessage
+from resolve_customer.entitlements import AWSCustomerEntitlement
+
+logger = logging.getLogger('sqs_event_manager')
+logger.setLevel('INFO')
+
+
+def lambda_handler(event, context):
+    """
+    Expects event type matching the SNS topic
+
+    {
+        'Records': [
+            {
+                'message_id': 'str',
+                'receipt_handle': 'str',
+                'body': {
+                    'Type' : 'str',
+                    'MessageId : 'str',
+                    'TopicArn' : 'str',
+                    'Message' : {
+                        'action': 'str',
+                        'customer-identifier': 'str',
+                        'product-code': 'str',
+                    },
+                    'Timestamp' : 'str',
+                    'SignatureVersion' : 'str',
+                    'Signature' : 'str',
+                    'SigningCertURL' : 'str',
+                    'UnsubscribeURL' : 'str'
+                },
+                'attributes': {
+                    'ApproximateReceiveCount': 'str',
+                    'SentTimestamp': 'str',
+                    'SenderId': 'str',
+                    'ApproximateFirstReceiveTimestamp': 'str'
+                },
+                'messageAttributes': {},
+                'md5OfBody': 'str',
+                'eventSource': 'aws:sqs',
+                'eventSourceARN': 'arn:aws:sqs:us-east-1:111122223333:my-queue',
+                'awsRegion': 'us-east-1'
+            }
+        ]
+    }
+    """
+    batch_item_failures = []
+    sqs_batch_response = {}
+
+    try:
+        for message in event['Records']:
+            process_message(message, batch_item_failures)
+    except Exception as error:
+        return json.dumps(
+            error_response(500, f'{type(error).__name__}: {error}')
+        )
+
+    sqs_batch_response['batchItemFailures'] = batch_item_failures
+    return json.dumps(sqs_batch_response)
+
+
+def process_message(record: dict, batch_item_failures: dict):
+    """
+    Handle message received from SQS queue
+
+    Ensure the message is proper format. Get the customer entitlements and
+    send the information to the SCC service.
+    """
+    try:
+        message = AWSSNSMessage(record)
+
+        if not message.action:
+            logger.info(f'Received an unknown message: {json.dumps(record)}')
+        elif message.action == 'entitlement-updated':
+            # Notify SCC of customer entitlement change for the given product
+            entitlements = AWSCustomerEntitlement(
+                message.customer_id,
+                message.product_code
+            )
+            if entitlements.error:
+                return error_response(400, entitlements.error)
+
+            request = {  # noqa TODO: Cleanup noqa
+                'isBase64Encoded': False,
+                'statusCode': 200,
+                'body': {
+                    'CustomerIdentifier': message.customer_id,
```

Comment:
@schaefi The resolve customer API returns both a customer id and an account number. Are these both needed by SCC? If so the account number is not readily available just from calling the get entitlements API.

Reply:
>  Are these both needed by SCC?

I don't know, we need to ask the SCC people tomorrow

> If so the account number is not readily available just from calling the get entitlements API.

I combined the output of both calls resolve_customer() and get_entitlements(). The CustomerAWSAccountId is expected to be part of the data from the resolve_customer() API call. The list of entitlements is the result from get_entitlements() for this customer ID

I can't tell if all information is present and correctly requested as I still was not able to test this code for real as I said.
I'm sure we will add some changes when we see the requests for real and also when we know what SCC wants.

Reply:
regarding customer id's I was reading this docs:
* https://docs.aws.amazon.com/marketplace/latest/APIReference/API_marketplace-metering_ResolveCustomer.html#AWSMarketplaceService-marketplace-metering_ResolveCustomer-response-CustomerAWSAccountId

which part SCC needs is still unknown to me, which is the reason why I added everything to the result so far

Reply:
plus the entitlements filtered by the CustomerID

Reply:
Yeah, it seems like entitlements are tied to customer id. So I think that should be enough 🤔